File: 品味Java的21種設計模式/dp_cpp/chainofresponsibility/example5/RequestModel.cpp
#include "RequestModel.h"

namespace cn {
namespace javass {
namespace dp {
namespace chainofresponsibility {
namespace example5 {

RequestModel::RequestModel(std::string type) {
    this->type = type;
}

std::string RequestModel::getType() {
    return type;
}
}
}
}
}
}

File: 品味Java的21種設計模式/dp_java/chainofresponsibility/example5/RequestModel.java
package cn.javass.dp.chainofresponsibility.example5;
/**
 * 通用的请求对象
 */
public class RequestModel {
    /**
     * 表示具体的业务类型
     */
    private String type;
    /**
     * 通过构造方法把具体的业务类型传递进来
     * @param type 具体的业务类型
     */
    public RequestModel(String type) {
        this.type = type;
    }
    public String getType() {
        return type;
    }
}